torch/nn/parallel/distributed.py

```diff
@@ -1060,6 +1060,39 @@ def _register_comm_hook(self, state: object, hook: callable):
         self._check_comm_hook(hook)
         dist._register_comm_hook(self.reducer, state, hook)
 
+    def _register_builtin_comm_hook(
+        self, comm_hook_type: dist.BuiltinCommHookType
+    ):
+        r"""
+        Registers a built-in communication hook that specifies how DDP
+        aggregates gradients across multiple workers.
```

Comment:
Let's mention here that the purpose of the builtin hooks are to provide efficient C++ implementations for certain hooks which might not be as efficient if implemented in python using a python communication hook.